lib/memstore.ts

```diff
@@ -188,12 +205,23 @@ export class MemoryCookieStore extends Store {
       return promiseCallback.promise
     }
 
-    const domainEntry: { [key: string]: any } =
-      this.idx[domain] ?? Object.create(null)
+    const domainEntry: {
+      [path: string]: {
+        [key: string]: Cookie
+      }
+    } =
+      this.idx[domain] ??
+      (Object.create(null) as {
+        [path: string]: {
+          [key: string]: Cookie
+        }
+      })
```

Comment:
```suggestion
      Object.create(null)
```
Pretty sure the type assertion shouldn't be necessary. But if it is, use `MemoryCookieStoreIndex[string]` instead.